ast/annotations.go

```diff
@@ -556,7 +556,11 @@ func attachAnnotationsNodes(mod *Module) Errors {
 		if a.Scope == "" {
 			switch a.node.(type) {
 			case *Rule:
-				a.Scope = annotationScopeRule
+				if a.Entrypoint {
```

Comment:
So the `document` scope is only implied if `entrypoint` is explicitly set to `true`?

```
# METADATA
# entrypoint: false
# custom:
#   only_applicable_to_doc_scope: I hope this doesn't show up in my rule annotations
allow if {
  ...
}
```

Only going by the docs, someone authoring the above would probably expect the metadata to be scoped to `document`. We should either make this explicit in the docs, or enforce the `document` scope regardless of its value (make it a pointer to a bool, so we can check for `nil`? 🤔).

Reply:
I can add to the docs that it applies when `entrypoint` is set to `true`. 

Changing to a pointer is more of a breaking change than anything in here so far, IMHO :)